File: backend/MillionProperty.Domain/Entities/Property.cs
namespace MillionProperty.Domain.Entities;

public class Property
{
    [BsonId] 
    [BsonRepresentation(BsonType.ObjectId)] 
    public string IdProperty { get; set; }= string.Empty; 

    public string Name { get; set; }= string.Empty; 
    public string Address { get; set; }= string.Empty; 
    public decimal Price { get; set; }
    public string CodeInternal { get; set; }= string.Empty; 
    public int Year { get; set; }

    public string Description { get; set; } = string.Empty; 
    public int Bedrooms { get; set; } 
    public int Bathrooms { get; set; } 
    public double SquareMeters { get; set; }

    [BsonRepresentation(BsonType.ObjectId)]
    public string IdOwner { get; set; } = string.Empty; 
}

File: frontend/src/types/property.ts
import { OwnerDetail, OwnerSummary } from "./owner";

export interface PropertyListItem {
  idProperty: string;
  name: string;
  address: string;
  price: number;
  imageUrl?: string | null;
  bedrooms: number;
  bathrooms: number;
  squareMeters: number;
  owner?: OwnerSummary | null;
}

export interface PropertyTrace {
  dateSale: string;
  name: string;
  value: number;
  tax: number;
}

export interface PropertyDetail {
  idProperty: string;
  name: string;
  address: string;
  price: number;
  codeInternal: string;
  year: number;
  description: string;
  bedrooms: number;
  bathrooms: number;
  squareMeters: number;
  owner?: OwnerDetail | null;
  imageUrls: string[];
  traces: PropertyTrace[];
}
